java-checks/src/main/java/org/sonar/java/checks/CognitiveComplexityMethodCheck.java

```diff
@@ -0,0 +1,476 @@
+
+      ExpressionTree left = binTree.leftOperand();
+      ExpressionTree right = binTree.rightOperand();
+
+      if (binTree.kind() != left.kind() && isLogicalOp(left)) {
+        total++;
+        addSecondaryLocation(flow, binTree, 1, 0);
+      }
+      if (binTree.kind() != right.kind() && isLogicalOp(right)) {
+        total++;
+        addSecondaryLocation(flow, binTree, 1, 0);
+      }
+      if (binTree.parent().is(CONDITIONAL_OR) && right.is(CONDITIONAL_AND)) {
+        total++;
+        addSecondaryLocation(flow, binTree, 1, 0);
+      }
+      expTree =  binTree.leftOperand();
+    }
+
+    return total;
+  }
+
+  private BinaryExpressionTree scootLeft(BinaryExpressionTree binaryExpressionTree) {
+
+    BinaryExpressionTree binTree = binaryExpressionTree;
+
+    if (binTree.is(CONDITIONAL_AND)) {
+      while (binTree.leftOperand().is(CONDITIONAL_AND)) {
+        binTree = (BinaryExpressionTree) binTree.leftOperand();
+      }
+    } else if (binTree.is(CONDITIONAL_OR)) {
+      ExpressionTree left = binTree.leftOperand();
+      while (left.is(CONDITIONAL_OR) && ! isLogicalOp(((BinaryExpressionTree)left).rightOperand())) {
+        binTree = (BinaryExpressionTree) binTree.leftOperand();
+        left = binTree.leftOperand();
+      }
+    }
+    return binTree;
+  }
+
+
+  private boolean isLogicalOp(ExpressionTree expTree) {
+    return expTree.is(CONDITIONAL_AND) || expTree.is(CONDITIONAL_OR);
+  }
+
+  private int countTryChains(Tree st, List<JavaFileScannerContext.Location> flow, int nestingLevel) {
+    int total = 0;
+
+    if (st.is(Tree.Kind.TRY_STATEMENT)) {
+      TryStatementTree tryStatment = (TryStatementTree) st;
+
+      List<CatchTree> catches = tryStatment.catches();
+      for (CatchTree catchTree : catches) {
+        total++;
+        total += nestingLevel;
+        addSecondaryLocation(flow, catchTree, total, nestingLevel);
+        total += countComplexity(catchTree.block().body(), flow, nestingLevel +1);
+      }
+
+      if (tryStatment.finallyBlock() != null && tryStatment.finallyBlock().body() != null) {
+        total += countComplexity(tryStatment.finallyBlock().body(), flow, nestingLevel);
+      }
+    }
+
+    return total;
+  }
+
+  private int countBreakAndContinue(Tree st, List<JavaFileScannerContext.Location> flow) {
+    if ( (st.is(BREAK_STATEMENT) && ((BreakStatementTree) st).label() != null)
+            || (st.is(CONTINUE_STATEMENT) && ((ContinueStatementTree) st).label() != null) ) {
+      flow.add(new JavaFileScannerContext.Location("+1", st));
+      return 1;
+
+    }
+
+    return 0;
+  }
+
+  private Tree getSegmentOfInterest(StatementTree st) {
+    switch (st.kind()) {
+      case LABELED_STATEMENT:
+        return ((LabeledStatementTree)st).statement();
+      case VARIABLE:
+        VariableTree vt = (VariableTree)st;
+        if (vt.initializer() != null) {
+          return vt.initializer();
+        }
+        break;
+      case EXPRESSION_STATEMENT:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1151)